lib/pure/collections/sequtils.nim

```diff
@@ -509,12 +509,37 @@ proc keepIf*[T](s: var seq[T], pred: proc(x: T): bool {.closure.})
       inc(pos)
   setLen(s, pos)
 
-func delete*[T](s: var seq[T]; first, last: Natural) =
+func delete*[T](s: var seq[T]; slice: Slice[int]) =
+  ## Deletes the items `s[slice]`, raising a defect if the slice contains
+  ## elements out of range.
+  runnableExamples:
+    var a = @[10, 11, 12, 13, 14]
+    doAssertRaises(AssertionDefect): a.delete(4..5)
+    assert a == @[10, 11, 12, 13, 14]
+    a.delete(4..4)
+    assert a == @[10, 11, 12, 13]
+    a.delete(1..2)
+    assert a == @[10, 13]
+    a.delete(1..<1) # empty slice
+    assert a == @[10, 13]
+  assert slice.a < s.len and slice.a >= 0 and slice.b < s.len, $(slice, s.len)
```

Comment:
```suggestion
  assert slice.a < s.len and slice.a >= 0 and slice.b < s.len, $(slice, s.len)
  if slice.b < slice.a: return
```
This saves unnecessary calculations and `newLen` would be wrong otherwise (e.g. for `2..0`, `newLen` would be `len(s) - 1`).

Reply:
done, slightly differently